airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -13,6 +16,49 @@ const getDefaultCursorField = (streamNode: SyncSchemaStream): string[] => {
   return streamNode.config?.cursorField || [];
 };
 
+const clearBreakingFieldChanges = (nodeStream: SyncSchemaStream, breakingChangesByStream: StreamTransform[]) => {
+  if (!breakingChangesByStream?.length) {
+    return nodeStream;
+  }
+
+  let clearPrimaryKey = false;
+  let clearCursorField = false;
+
+  for (const streamTransformation of breakingChangesByStream) {
+    // get all of the removed field paths for this transformation
+    const removedFieldPaths = streamTransformation.updateStream?.map((update) => update.fieldName);
+
+    if (!removedFieldPaths?.length) {
+      return nodeStream;
+    }
+
+    // if there is a primary key in the config, and any of its field paths were removed, we'll be clearing it
+    if (
```

Comment:
We _only_ want to be overriding this if true, hence why it makes sense to use an if statement here rather than setting `clearPrimaryKey` as a boolean of this statement.  Otherwise we end up with `clearPrimaryKey = !!(!!nodeStream.config?.primaryKey?.length && ...` and things get less readable.